src/onnx/parse_gelu.cpp

```diff
@@ -0,0 +1,116 @@
+}
+
+instruction_ref parse_gelu_tanh_approx(const onnx_parser::node_info& info, instruction_ref x)
+{
+    auto x_lens     = x->get_shape().lens();
+    auto x_type     = x->get_shape().type();
+    auto fit_const  = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {0.044715f}});
+    auto sqrt_2_rpi = info.add_literal(
+        migraphx::literal{migraphx::shape{x_type}, {static_cast<float>(sqrt(M_2_PI))}});
+    auto one   = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {1.0f}});
+    auto half  = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {0.5f}});
+    auto three = info.add_literal(migraphx::literal{migraphx::shape{x_type}, {3.0f}});
+    auto three_mbcast =
+        info.add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), three);
+    auto pow0             = info.add_instruction(migraphx::make_op("pow"), {x, three_mbcast});
+    auto fit_const_mbcast = info.add_instruction(
+        migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), fit_const);
+    auto mul0 = info.add_instruction(migraphx::make_op("mul"), {pow0, fit_const_mbcast});
+    auto add0 = info.add_instruction(migraphx::make_op("add"), {mul0, x});
+    auto sqrt_2_rpi_mbcast = info.add_instruction(
+        migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), sqrt_2_rpi);
+    auto mul1  = info.add_instruction(migraphx::make_op("mul"), {add0, sqrt_2_rpi_mbcast});
+    auto tanh0 = info.add_instruction(migraphx::make_op("tanh"), mul1);
+    auto one_mbcast =
+        info.add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), one);
+    auto add1 = info.add_instruction(migraphx::make_op("add"), {tanh0, one_mbcast});
+    auto half_mbcast =
+        info.add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), half);
+    auto mul2 = info.add_instruction(migraphx::make_op("mul"), {x, half_mbcast});
+    return info.add_instruction(migraphx::make_op("mul"), {add1, mul2});
+}
+
+struct parse_gelu : op_parser<parse_gelu>
+{
+    std::vector<op_desc> operators() const { return {{"Gelu"}}; }
+    instruction_ref parse(const op_desc& /*opd*/,
+                          const onnx_parser& /*parser*/,
+                          onnx_parser::node_info info,
+                          std::vector<instruction_ref> args) const
+    {
+        std::string approximate = "none";
+        auto x                  = args[0];
+        if(not is_type_float(x->get_shape().type()))
+        {
+            MIGRAPHX_THROW("PAERSE_GELU: input tensor is not a floating type");
+        }
+
+        if(contains(info.attributes, "approximate"))
+        {
+            approximate = info.attributes["approximate"].s();
+        }
+        if(approximate == "tanh")
+        {
+            return parse_gelu_tanh_approx(info, x);
+        }
+        else
+        {
+            return parse_gelu_no_approx(info, x);
```

Comment:
We should just insert the `erf` version even if it sets to `tanh`. `rewrite_gelu` will rewrite it to use tanh as needed.

Reply:
Can't this Gelu be inserted just as a 'gelu' operator, and then rewrite_gelu could insert the appropriate code..? Thanks.

Reply:
We should avoid creating operators that are just a composite of other operators. It easy enough to match the erf version(and we already have a matcher for it).